### PR TITLE
[WIP] merging eslint files

### PR DESCRIPTION
**Pull request by rsun19**
[RHOAIENG-2534](https://issues.redhat.com/browse/RHOAIENG-2534)

## Description

merging eslint config files

## How Has This Been Tested?

Break some rules in various eslint files. It should show, and `npm run test` should work in root, backend, and frontend directories.

## Test Impact

N/A

## Request review criteria:

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Commits have been squashed into descriptive, self-contained units of work (e.g. 'WIP' and 'Implements feedback' style messages have been removed)
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has added tests or explained why testing cannot be added (unit or cypress tests for related changes)

If you have UI changes: 

- [ ] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Included tags to the UX team if it was a UI/UX change.

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -51,5 +52,22 @@
     "dotenv-expand": "^5.1.0",
     "npm-run-all": "^4.1.5",
     "prettier": "^2.2.1"
+  },
+  "devDependencies": {
+    "@typescript-eslint/eslint-plugin": "^7.1.1",
```
Comment (rsun19) [inline package.json:57]: For some reason node version `18.18.0` is needed :(

Comment (rsun19) [reply package.json]: resolved by moving to optionalDependencies

File: frontend/tsconfig.json [hunk 2]
```diff
@@ -17,6 +17,7 @@
     "esModuleInterop": true,
     "allowSyntheticDefaultImports": true,
     "strict": true,
+    "strictNullChecks": true,
```
Comment (christianvogt) [inline frontend/tsconfig.json:20]: This is redundant because we use `strict: true`

File: .eslintrc [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+    // "jsx-a11y/no-autofocus": ["error", { "ignoreNonDOM": true }],
+    // "jsx-a11y/anchor-is-valid": [
+    //   "error",
+    //   {
+    //     "components": ["Link"],
+    //     "specialLink": ["to"],
+    //     "aspects": ["noHref", "invalidHref", "preferButton"]
+    //   }
+    // ],
+    // "arrow-body-style": "error",
```
Comment (christianvogt) [inline .eslintrc:77]: Instead of commenting out all these rules, move the common rules that we want to support from the frontend to the root. Then in the backend add the same rule as disabled if it's causing issues an we can address them later.

But for a first step this is ok.

File: frontend/src/concepts/pipelines/topology/usePipelineTaskTopology.tsx [hunk 4]
```diff
@@ -152,8 +152,10 @@ const getNodesForTasks = (
         (e) =>
           e.getCustomPropertiesMap().get('task_name')?.getStringValue() === (taskName || taskId),
       );
-      if (execution) {
-        linkedArtifacts = executionLinkedArtifactMap[execution.getId()] ?? [];
+      if (execution && execution.getId() in executionLinkedArtifactMap) {
+        linkedArtifacts = executionLinkedArtifactMap[execution.getId()];
+      } else {
+        linkedArtifacts = [];
```
Comment (christianvogt) [inline frontend/src/concepts/pipelines/topology/usePipelineTaskTopology.tsx:158]: Why the change?

File: .eslintrc [hunk 5]
```diff
@@ -0,0 +1,224 @@
...
+    "plugin:prettier/recommended",
+    "prettier"
+  ],
+  "ignorePatterns": [
+    "package.json",
+    "./frontend/src/third_party/mlmd",
+    "./frontend/src/__tests__/cypress/.nyc_output",
+    "./frontend/src/__tests__/cypress/coverage",
+    "./frontend/package.json",
+    "./frontend/config"
```
Comment (christianvogt) [inline .eslintrc:28]: Can these stay in frontend or does it end up overriding the root level ignore patterns?

**Comment by rsun19** [comment]
@christianvogt made requested changes. For this to work, we will need to notify people to get rid of their working directory eslint setting, since our combined eslintrc file seems to confuse the vscode eslint extension.
